mutations_test.go

```diff
@@ -310,6 +310,94 @@ func TestImageSkaffold(t *testing.T) {
 	assert.Equal(t, "docker.sqooba.io/sqooba/sqooba-website:32cbc804-dirty@sha256:a4a729d8691ed70eb56cf03053333cf42e8a6c33f6ee67ea862da4459d7f70fd", patches[0].Value)
 }
 
+func TestImageInternalFullRegistry(t *testing.T) {
+
+	wh := mutationWH{
+		registry: "docker.sqooba.io/public-docker-virtual",
+	}
+
+	pod := corev1.Pod{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{Image: "docker.sqooba.io/local-repo/xyz/image:snapshot"},
+			},
+		},
+	}
+
+	patches, err := wh.applyMutationOnPod(pod)
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(patches))
+	assert.Equal(t, "replace", patches[0].Op)
+	assert.Equal(t, "/spec/containers/0/image", patches[0].Path)
+	// actual behaviour if the registry docker.sqooba.io/local-repo is not ignored
+	assert.Equal(t, "docker.sqooba.io/public-docker-virtual/local-repo/xyz/image:snapshot", patches[0].Value)
+}
+
+func TestImageInternalFullRegistryWithIgnore(t *testing.T) {
```

Comment:
I'd ensure that anything different than `docker.sqooba.io/local-repo` is still replaced with `docker.sqooba.io/public-docker-virtual`